Separate address fields

stijnjanssens opened the pull request:
Added the option to separate address fields in Street + nr, Zip and City
Looking forward to your feedback!

stijnjanssenstd committed:
Separate address fields

Added the option to separate address fields in Street + nr, Zip and City

AugustMiller commented:
Wow! This is awesome. Looked over the source briefly, but haven't tried it out in an active project.

Thanks so much for taking the time to contribute. I'll do my best to review early next week.

🎄

AugustMiller commented:
Any reason we couldn't include a few more standard address fields?

- [ ] Address
- [ ] Address 2
- [ ] City
- [ ] State/Province
- [ ] ZIP
- [ ] Country

Might be worth abstracting some of this into an `addressField` method, so as to reduce the amount of boilerplate for all these.

Mostly asking, because of the framework you've set up (joining various parts of a query) makes it fairly extensible—seems like the right time to tackle this, which has been on my list for a while (saving complete addresses). Thoughts on how this works with plain searches (i.e. _Eiffel Tower_)?

This would probably be backwards-compatible, given that the text portion is already stored as `address`, we could easily add `address2`, `city`, `state`, etc…

Thanks again—looking forward to your thoughts!

stijnjanssenstd committed:
Fields option added

- Replace the ‘separate’ option with a ‘fields’ option so that the user can choose wich fields he would like to store.
- some code cleanup

stijnjanssens commented:
Hi August, thanks for the enthusiastic response! I've taken your feedback into account and came up with the idea to let the DEV choose which separate fields he would like to store via blueprint:

```yml
restoinfo_address:
  label: Address
  type: map
  required: true
  fields:
    address: Street + Number
    address2: Address 2
    zip: Zipcode
    city: City
    country: Country
  center:
    lat: 50.8685716
    lng: 4.40442319999
    zoom: 15
  help: Move the pin wherever you'd like, or search for a location!
```

In this way we can store as many fields as required by the project.
By default (if the fields option is not set) we fall back to the original field ('address')

Searches like "Eifel tower" still work, although more as a result of google's flexibility ;)
At the moment there is no validation or specific required parts/fields possible.

I've used `address_field` as a method to add the different inputs to the container, feedback welcome!

AugustMiller commented:
I think I'd prefer to keep this as simple as possible for the implementing developer, rather than pushing address/formatting boilerplate down into the YAML (and effectively opening it up to fields of any name). Strictly as far as support and documentation are concerned, I'd rather have a couple of sane defaults than unlimited possibilities.

Instead, we may want to include something like `format` which defaults to `search` and allows for one or two predetermined keys (say, `address` and `international` or something), which show a corresponding set of fields. In theory, this selection of fields could be localized, too?

This way, we'll also retain control of the fields' layout.

```yml
restoinfo_address:
  label: Address
  type: map
  required: true
  format: search # <- Provides just the search/keyword field (default)
  center:
    lat: 50.8685716
    lng: 4.40442319999
    zoom: 15
  help: Move the pin wherever you'd like, or search for a location!
```

…and, alternatively…

```yml
  format: address # <- Complete address fields
```

stijnjanssens commented:
It is a good idea to limit the options of the dev to choose from. I think
that's a good balance between flexibility and control. We can foresee some
fields (address, address2, street, number, box, state, province, zip,
country) and it is up-to the dev to choose which he wants to use...
It's pretty hard to determine a configuration that fits all use cases,
sometimes you want the number separate from the street and box, sometimes
you don't.

Proposal for the other option, predefined formats:
search (default)
-- address

national
-- address
-- zip
-- city

international
-- address
-- address2
-- city
-- state/province
-- zip
-- country

I don't know if the name "national" vs "international" is clear enough...

Yet another options could be to make it possible to exclude certain fields,
but this could be kind of confusing, counter intuitive...

 What do you think?
Stijn

2017-12-19 20:07 GMT+01:00 August Miller <notifications@github.com>:

> I think I'd prefer to keep this as simple as possible for the implementing
> developer, rather than pushing address/formatting boilerplate down into the
> YAML (and effectively opening it up to fields of any name). Strictly as far
> as support and documentation are concerned, I'd rather have a couple of
> sane defaults than unlimited possibilities.
>
> Instead, we may want to include something like format which defaults to
> search and allows for one or two predetermined keys (say, address and
> international or something), which show a corresponding set of fields. In
> theory, this selection of fields could be localized, too?
>
> This way, we'll also retain control of the fields' layout.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/AugustMiller/kirby-map-field/pull/16#issuecomment-352856356>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AHT6HiPqa57r8UykP5pLUXAI3oKrii2Hks5tCAl2gaJpZM4REd3J>
> .
>
